packages/fluentui/react/src/components/Flex/FlexItem.tsx

```diff
@@ -113,3 +67,81 @@ const applyStyles = (element: React.ReactElement<any>, styles, classes): React.R
     styles: mergeStyles(styles.root || {}, element.props.styles)
   });
 };
+
+/**
+ * A FlexItem is a layout component that customizes alignment of Flex child.
+ */
+const FlexItem: React.FC<FlexItemProps> & { className: string; __isFlexItem: boolean } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(FlexItem.displayName, context.telemetry);
+  setStart();
+
+  const { align, children, className, design, grow, flexDirection, push, shrink, size, styles, variables } = props;
+
+  const { classes, styles: resolvedStyles } = useStyles<FlexItemStylesProps>(FlexItem.displayName, {
+    className: FlexItem.className,
+    mapPropsToStyles: () => ({
+      align,
+      grow,
+      flexDirection,
+      push,
+      shrink,
+      size
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables
+    }),
+    rtl: context.rtl
+  });
+
+  let element: React.ReactElement;
+
+  // pass calculated bits using Render Props pattern
+  if (typeof children === 'function') {
+    element = children({
+      styles: resolvedStyles.root,
+      classes: classes.root
+    });
+  } else if (_.isNil(children)) {
+    element = null;
+  } else {
+    element = applyStyles(React.Children.only(children) as React.ReactElement, resolvedStyles, classes);
+  }
+
+  setEnd();
+
+  return element;
+};
+
+FlexItem.className = 'ui-flex__item';
+FlexItem.displayName = 'FlexItem';
+
+FlexItem.propTypes = {
+  ...commonPropTypes.createCommon({
+    children: false,
+    accessibility: false,
+    content: false
+  }),
+  children: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+
+  align: PropTypes.oneOf(['auto', 'start', 'end', 'center', 'baseline', 'stretch']),
+  size: PropTypes.oneOfType([PropTypes.oneOf(['size.half', 'size.quarter', 'size.small', 'size.medium', 'size.large']), PropTypes.string]),
```

Comment:
A bit worried that this can be any css value (px or %) which may cause problems with the caching... :(

Reply:
Let's see the feedback. I agree there that it can be a real issue, but:
- I don't think that it's used enough frequently
- `size` manages `flexBasis`, may be in this case we should manage it via `design` prop...
- we can disable caching for `FlexItem`